apisix/plugins/ext-plugin/init.lua

```diff
@@ -400,9 +442,26 @@ local rpc_handlers = {
             return nil, "failed to send RPC_HTTP_REQ_CALL: " .. err
         end
 
-        local ty, resp = receive(sock)
-        if ty == nil then
-            return nil, "failed to receive RPC_HTTP_REQ_CALL: " .. resp
+        local ty, resp
+        while true do
+            ty, resp = receive(sock)
+            if ty == nil then
+                return nil, "failed to receive RPC_HTTP_REQ_CALL: " .. resp
+            end
+
+            if ty ~= constants.RPC_EXTRA_INFO then
+                break
+            end
+
+            local out, err = handle_extra_info(ctx, resp)
+            if not out then
+                return nil, "failed to handle RPC_EXTRA_INFO: " .. err
+            end
+
+            local ok, err = send(sock, constants.RPC_EXTRA_INFO, out)
+            if not ok then
+                return nil, "failed to reply RPC_EXTRA_INFO: " .. err
+            end
         end
```

Comment:
Can you explain the relationship between `RPC_EXTRA_INFO` and `RPC_HTTP_REQ_CALL`? I don't understand too much. Why do I receive an `RPC_EXTRA_INFO` response when I send an `RPC_HTTP_REQ_CALL` request? What is his application scenario?

Reply:
The detail can be found in https://lists.apache.org/thread.html/r03232d5a48ba1d71c1b62185894265348abbf3bede13fba7edd38255%40%3Cdev.apisix.apache.org%3E.

Quoted from there:
```
APISIX sends HTTPReqCallReq
while reply from Runner isn't HTTPReqCallResp:
    APISIX handle ExtraInfoReq
    APISIX sends ExtraInfoResp
APISIX handle HTTPReqCallResp
```

Reply:
OK, I see